src/test/java/org/broadinstitute/hellbender/utils/config/ConfigIntegrationTest.java

```diff
@@ -0,0 +1,126 @@
+import java.io.FileInputStream;
+import java.lang.reflect.Method;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.Properties;
+
+/**
+ * Integration test
+ * Created by jonn on 12/7/17.
+ */
+public class ConfigIntegrationTest extends CommandLineProgramTest {
+
+    @CommandLineProgramProperties(
+            summary = "Dummy empty command line that requires a reference .",
+            oneLineSummary = "empty class",
+            programGroup = TestProgramGroup.class
+    )
+    public static class DummyGatkTool extends GATKTool {
+
+        @Argument(
+                doc = "Output file with only the configuration settings in it.",
+                fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+                shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME
+        )
+        private File outputFilePath;
+
+        @Override
+        public void traverse() {
+            final GATKConfig gatkConfig = ConfigFactory.getInstance().getOrCreate(GATKConfig.class);
+            ConfigFactory.dumpConfigSettings( gatkConfig, outputFilePath.toPath() );
+        }
+    }
+
+    @Override
+    public String getTestedClassName() {
+        return DummyGatkTool.class.getSimpleName();
+    }
+
+    @DataProvider
+    private Object[][] provideForTestToolWithConfigValidation() {
+        return new Object[][] {
+                { getSafeNonExistentPath("nonExistentConfig.config") },
+        };
+    }
+
+    @Test(dataProvider = "provideForTestToolWithConfigValidation")
+    public void testToolWithConfigValidation(final Path configFilePath) {
+        // From droazen:
+//      A better integration test, I think, would be to create a private static dummy tool
+//      class here that just dumps selected items from its configuration to a file in apply(), run that dummy tool
+//      in your test, and assert that the configuration dumped is consistent with the one specified on the command line.
+//
+//      You could have two test cases: one with the default config file, and one with a user-specified config file
+//      that alters one or more default settings.
+
+        final File tmpFile = getSafeNonExistentFile("ConfigIntegrationTest.output.txt");
+        tmpFile.deleteOnExit();
+
+        // Create some arguments for our command:
+        final ArgumentsBuilder args = new ArgumentsBuilder();
+        args.add("--" + StandardArgumentDefinitions.GATK_CONFIG_FILE_OPTION);
+        args.add(configFilePath.toUri().toString());
+        args.add("-" + StandardArgumentDefinitions.OUTPUT_SHORT_NAME);
+        args.add(tmpFile);
+
+        // Run our command:
+        runCommandLine(args.getArgsArray());
+
+        // Now we get to read in the file's contents and compare them to our config settings:
+        final Properties properties = new Properties();
+        try ( final FileInputStream inputStream = new FileInputStream(tmpFile) ) {
+            properties.load(inputStream);
```

Comment:
You do not appear to ever use the properties that you load here, and which were output by the test tool, including in your assert below! This suggests that this test is not actually checking the output of `DummyGatkTool` at all.

Reply:
Wow.  I guess I lost my train of thought somewhere in there.

Fixed.